virtue/virtue-base/kickoff_root.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# This script is ran by the root user; add any necessary
+# root commands that must be ran during startup such as
+# adding iptable rules.
+
+# Allow enough time for the network rules to be copied over
+sleep 10
+
+# Loop through the networkRules file and add them to the
+# iptable rule chains
+cat /etc/networkRules | while read line; do
+	iptables $line
+done
+
+
+# Since the virtue user has a `nologin` shell, we cannot simply
+# `su` to virtue. Instead we execute the script on behalf of the
+# virtue user.
+su -s /bin/sh -c '/home/virtue/kickoff.sh' virtue
```

Comment:
According to some brief Googling, it doesn't look like `su` is a good choice for dropping privs, especially if we end up somewhere with PAM enabled. There are some other wrapper tools that might be useful, e.g. those mentioned in https://unix.stackexchange.com/questions/132663/how-do-i-drop-root-privileges-in-shell-scripts

Reply:
Great find, the script now uses **runuser** which is supposed to be safe.